packages/accounts-controller/src/AccountsController.ts

```diff
@@ -731,25 +731,25 @@ export class AccountsController extends BaseController<
 
       // handle if the selected account was deleted
       if (!this.getAccount(this.state.internalAccounts.selectedAccount)) {
```

Comment:
It looks like the selected account is already being updated in this block. So now it's being updated twice redundantly.

Could you elaborate further on what this was intended to accomplish?

Reply:
I see, it was to ensure that the account was removed in the first update. That makes sense, though surely we can refactor this to avoid making the same state update again redundantly.

Reply:
The onKeyringStateChange has been changed to only update once at the end. 3f27d6476f26b52f60cd53cfdbd17ea6012cefee